storage/src/index/storage.rs

```diff
@@ -18,13 +18,25 @@ pub struct ValueIterator<'a, V> {
     next: Option<&'a Record<V>>,
 }
 
+// An iterator over all values associated with a translated key, allowing for mutation.
+pub struct MutableValueIterator<'a, V> {
```

Comment:
(As discussed, would be interesting to add "deletion" to this somehow and/or to combine get/put)

Reply:
Not seeing a straightforward way to accomplish this since with the linked list you're only able to mutably refer to a single entry at a time.